### ISSUE TITLE
Should not filter out "default" audio device

### ISSUE THREAD

**Issue by davidzhao** [livekit/client-sdk-js#483]
### Describe the problem

Currently we are filtering out the `default` device from [DeviceManager.getDevices](https://github.com/livekit/client-sdk-js/blob/main/src/room/DeviceManager.ts#L71)

The intention of this is to remove the duplicate device that shows up. However, this has an unintended consequence of not being able to react to system-wide device changes.

When a user change their system device from one input to another, we are not able to react to it automatically.

### Describe the proposed solution

Eliminate the filtering and ensure we honor "default" as a viable deviceId.

### Importance

would make my life easier

**Comment by salihsagdilek** [comment]
Hello @lukasIO @davidzhao, We see that the default device is on the list now, this is good news! However, when there is a new device connected, sometimes voice input is taken from the previous device. Even sometimes we see that sound comes robotic when default device changed.

**Comment by lukasIO** [comment]
hi @salihsagdilek 
thanks for the report, can you give us some more details on this?
do you see this happening (only) when switching from/to bluetooth devices?
on which browser are you seeing this?

**Comment by salihsagdilek** [comment]
Hi @lukasIO 
Experienced this on bluetooth device but didn’t find a chance to try it on a cable device. I use Chrome version 107.
Case 1
Steps to produce:
I join the room with an audio device (device 1) - My setting is “default mic and default speaker”
I connected another audio device (device 2).
What happens:
My default speaker goes to device 2 as expected (I hear others from device 2).
However, my audio still comes from device 1 (others hear me from device 1).
Case 2
Steps to produce
I have 2 devices connected (device 1 and device 2). Both my speaker and mic settings are on device 2 (this is an airpod - if that helps).
Then I disconnect device 2
What happens:
My settings (both speaker and mic) fall back to device 1 as expected
However, the audio starts being robotic. I attached a sample recording if that helps: https://vocaroo.com/1g7GJVLY8FhM
It also affects the other users in the room. Imagine there are 3 users in the room. When user1 changes the device settings and end up with robotic audio, user2 also starts sending robotic audio for some reason - even though nothing changed on user2. User 3 hears both users’ audio robotic.
